portal/eproms_substudy_tailored_content/src/js/Base.js

```diff
@@ -0,0 +1,467 @@
+
+export default {
+    created() {
+        PromiseAllSettledPolyfill();
+    },
+    mounted() {
+        Promise.allSettled([
+            this.$http(this.settingsURL).catch(error => { return error }),
+            this.$http(this.meURL).catch(error => { return error }),
+            this.$http("//geoip.cirg.washington.edu/json/"),
+            this.$http(`/static/js/src/data/common/empro_domain_mappings.json`),
+        ]).then(responses => {
+            try {
+                //set config settings
+                this.setSettings(JSON.parse(responses[0].value));
+            } catch(e) {
+                //log error to console
+                this.setErrorMessage(`Error parsing data ${e}`);
+            }
+            try {
+                //set user email and id
+                this.userInfo = JSON.parse(responses[1].value);
+            } catch(e) {
+                //log error to console
+                this.setErrorMessage(`Error parsing user data ${e}`);
+            }
+            try {
+                //set country code
+                this.setCountryCode(JSON.parse(responses[2].value));
+            } catch(e) {
+                //log error to console
+                this.setErrorMessage(`Error parsing country code data ${e}`);
+            }
+            try {
+                //set domain mappings
+                this.domainMappings = JSON.parse(responses[3].value);
+            } catch(e) {
+                //log error to console
+                this.setErrorMessage(`Error parsing domain mapping data ${e}`);
+            }
+            this.initApp();
+            //console.log("settings? ", this.settings);
+            //console.log("user? ", this.userInfo)
+        }).catch(error => {
+            this.setErrorMessage(`Error in promises ${error}`);
+            this.initApp();
+        });
+    },
+    watch: {
+        $route(to, from) {
+            /*
+             * watch for manual hash change
+             */
+            if (to && to.params) {
+                if (!to.params.topic) {
+                    location.reload();
+                    return false;
+                }
+                if (this.domains.indexOf(to.params.topic) !== -1) {
+                    location.reload();
+                    return false;
+                }
+            }
+        }
+    },
+    mixins: [NavMethods, VideoMethods],
+    /*
+     * methods available to the application
+     */
+    methods: {
+        isLoading() {
+            return this.loading;
+        },
+        isInitialized() {
+            return this.initialized;
+        },
+        initApp() {
+            this.setSettings(this.settings);
+            this.setLRBaseURL(this.settings?this.settings["LR_ORIGIN"]:"");
+            this.setUserID(this.userInfo?this.userInfo["id"]:0);
+            this.initialized = true;
+            Promise.allSettled(
+                this.getUserID()?
```

Comment:
if you only need the <user_id>, /api/me is much lighter weight, and i noticed you have a reference to that above...

Reply:
yes, will be using `/api/me` to get the user ID